system-contracts/contracts/ContractDeployer.sol

```diff
@@ -74,11 +77,12 @@ contract ContractDeployer is IContractDeployer, ISystemContract {
     function updateNonceOrdering(AccountNonceOrdering _nonceOrdering) external onlySystemCall {
         AccountInfo memory currentInfo = accountInfo[msg.sender];
 
-        require(
-            _nonceOrdering == AccountNonceOrdering.Arbitrary &&
-                currentInfo.nonceOrdering == AccountNonceOrdering.Sequential,
-            "It is only possible to change from sequential to arbitrary ordering"
-        );
+        if (
+            _nonceOrdering != AccountNonceOrdering.Arbitrary &&
```

Comment:
Should be `| |` instead of `&&` here.

Counterexample for mismatch between new and old code:
- _nonceOrdering != AccountNonceOrdering.Arbitrary
- currentInfo.nonceOrdering == AccountNonceOrdering.Sequential

Will fail in the old code and not in the current code.